File: LinkERP.BLL/SYS/CountryService.cs
using LinkERP.BLL.SYS.Interfaces;
using LinkERP.DAL.SYS.Interfaces;
using LinkERP.Entity.SYS;
using System;
using System.Collections.Generic;
using System.Text;
namespace LinkERP.BLL.SYS
{
    public class CountryService :ICountryService
    {
        ICountryRepository country ;
        public CountryService(ICountryRepository _country)
        {
            country = _country;
        }
        public string AddCountry(LBS_SYS_Country  lBS_SYS_Country)
        {
            return country.AddCountry(lBS_SYS_Country);
        }
        public string UpdateCountry(LBS_SYS_Country lBS_SYS_Country)
        {
            return country.UpdateCountry(lBS_SYS_Country);
        }
        public LBS_SYS_Country GetCountryByID(Guid CountryID)
        {
            return country.GetCountryByID(CountryID);
        }
        public bool DeleteCountryByID(Guid ID, string DeletedBy)
        {
            return country.DeleteCountryByID(ID, DeletedBy);
        }

    }
}

File: LinkERP/ClientApp/src/app/sys/services/country.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AppConfigService } from '../../app-config.service';

@Injectable({
  providedIn: 'root'
})
export class CountryService {
  private url: string = "";
  private baseurl: string = "";

 constructor(public http: HttpClient, private appConfigService: AppConfigService) {
    this.baseurl = appConfigService.getServiceBaseUrl();
 }

  getCountryByID(CountryID) {
    this.url = this.baseurl + 'api/SYS/Country/GetCountryByID/' + CountryID;
    return this.http.get(this.url);
  }
  addCountry(data) {
    this.url = this.baseurl + 'api/SYS/Country/AddCountry';
    return this.http.post(this.url, data);
  }
  updateCountry(data) {
    this.url = this.baseurl + 'api/SYS/Country/UpdateCountry';
    return this.http.post(this.url, data);
  }
  deleteCountryByID(CountryID, DeletedBy) {
    this.url = this.baseurl + 'api/SYS/Country/DeleteCountryByID/' + CountryID + '/' + DeletedBy;
    return this.http.post(this.url, null);
  }
}
